stdlib/CMakeLists.txt

```diff
@@ -44,81 +44,51 @@ if(SWIFT_BUILD_STATIC_STDLIB)
   list(APPEND SWIFT_STDLIB_LIBRARY_BUILD_TYPES STATIC)
 endif()
 
-add_custom_target(swift-stdlib-all)
-add_custom_target(swift-stdlib-sib-all)
-add_custom_target(swift-stdlib-sibopt-all)
-add_custom_target(swift-stdlib-sibgen-all)
-foreach(SDK ${SWIFT_SDKS})
-  add_custom_target("swift-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}")
-  add_custom_target("swift-test-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}")
-  add_custom_target("swift-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-sib")
-  add_custom_target("swift-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-sibopt")
-  add_custom_target("swift-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-sibgen")
-
-  set_property(TARGET
-      "swift-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}"
-      "swift-test-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}"
-      "swift-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-sib"
-      "swift-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-sibopt"
-      "swift-stdlib-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-sibgen"
-      PROPERTY FOLDER "Swift libraries/Aggregate")
-
-  foreach(ARCH ${SWIFT_SDK_${SDK}_ARCHITECTURES})
-    set(VARIANT_SUFFIX "-${SWIFT_SDK_${SDK}_LIB_SUBDIR}-${ARCH}")
-    add_custom_target("swift-stdlib${VARIANT_SUFFIX}")
-    add_custom_target("swift-stdlib${VARIANT_SUFFIX}-sib")
-    add_custom_target("swift-stdlib${VARIANT_SUFFIX}-sibopt")
-    add_custom_target("swift-stdlib${VARIANT_SUFFIX}-sibgen")
-    add_custom_target("swift-test-stdlib${VARIANT_SUFFIX}")
-
-    set_property(TARGET
-        "swift-stdlib${VARIANT_SUFFIX}"
-        "swift-stdlib${VARIANT_SUFFIX}-sib"
-        "swift-stdlib${VARIANT_SUFFIX}-sibopt"
-        "swift-stdlib${VARIANT_SUFFIX}-sibgen"
-        "swift-test-stdlib${VARIANT_SUFFIX}"
-        PROPERTY FOLDER "Swift libraries/Aggregate")
+function(define_stdlib_build_targets target suffix all_variant)
+  if(${all_variant})
```

Comment:
I think you can remove the `${…}`, which makes clear this is a boolean parameter. Personal preference, so feel free to ignore.

Below there’s a `if(NOT ${all_variant})` which I don’t know if you can do `if (NOT all_variant)`, but it might be worth a try.

Reply:
CMake `if` is weird in that it implicitly expands variables one level. It's probably a good habit to either *not* use `${…}` or to put the expansion in quotes.